c_api/Makefile

```diff
@@ -0,0 +1,107 @@
+# Makefile to aid with local development and testing
+# This is not required for automated builds
+
+# Detect architecture
+ARCH := $(shell uname -m)
+# Detect OS
+ifeq ($(OS),Windows_NT)
+	PLATFORM := win
+else
+	UNAME := $(shell uname)
+    ifeq ($(UNAME),Linux)
+        PLATFORM := linux
+    endif
+    ifeq ($(UNAME),Darwin)
+        PLATFORM := mac
+    endif
+endif
+
+TARGET_DIR := $(realpath ../target)
+
+# Helper function to create a zip file
+define make_zip
+    echo "Reading version from $(1)/c2pa.h"; \
+    if [ "$(PLATFORM)" = "win" ]; then \
+        VERSION=$$(findstr "^// Version:" $(1)/c2pa.h | awk -F": " "{print \$$2}"); \
+    else \
+        VERSION=$$(grep '^// Version:' $(1)/c2pa.h | sed 's/^\/\/ Version: //'); \
+    fi; \
+    ZIP_NAME=$(TARGET_DIR)/artifacts/c2pa-v$${VERSION}-$(2).zip; \
+    echo "Creating zip file: $${ZIP_NAME}"; \
+    mkdir -p $(TARGET_DIR)/artifacts $(1)/include $(1)/lib; \
+    cp $(1)/c2pa.h $(1)/include/; \
+    cp $(1)/libc2pa_c.* $(1)/lib/; \
+    cd $(1) && zip -r $${ZIP_NAME} include lib; \
+    echo "Zip file created: $${ZIP_NAME}"
+endef
+
+check-format:
+	cargo +nightly fmt -- --check
+
+clippy:
+	cargo clippy -- -D warnings
+
+test-local:
+	cargo test
+
+# Full local validation, build and test all features including wasm
+# Run this before pushing a PR to pre-validate
+test: check-format clippy test-local
+
+fmt:
+	cargo +nightly fmt
```

Comment:
Do we want to keep the `+nightly` in the check everywhere?

Reply:
We use nightly only for fmt because the nightly version includes some extra checking no available elsewhere.